src/client/graphics/layers/RadialMenu.ts

```diff
@@ -267,7 +267,36 @@ export class RadialMenu implements Layer {
   }
 
   tick() {
-    // Update logic if needed
+    // Only update when menu is visible
+    if (this.isVisible && this.clickedCell) {
```

Comment:
You can reduce the level of indentation by using the early exit strategy:
```suggestion
    if (!this.isVisible || this.clickedCell === null) return;
```